data-serving/data-service/src/controllers/case.ts

```diff
@@ -33,13 +34,16 @@ export const list = async (req: Request, res: Response): Promise<void> => {
         return;
     }
     // Filter query param looks like &q=some%20search%20query
-    const searchQuery = String(req.query.q || '').trim();
-    const query = searchQuery
+    if (typeof req.query.q !== 'string' && typeof req.query.q !== 'undefined') {
```

Comment:
Can these checks be done in the OpenAPI?

Reply:
We could, but we still have to do those here nonetheless, it's typescript types for req.query that is either string | string[] for any request params.

Reply:
Got it. We should add that indication to openAPI as well. We are using openAPI for verification and IIRC query is specified to be a string right now, so if a string[] was passed openAPI would fail validation.

Reply:
OpenAPI has this:
```
- name: q
          in: query
          description: The search query
          required: false
          schema:
            type: string
```

So this is correct, I don't know what you want to add to it?
If clients pass multiple &q= params to the URL that's their fault and yes it will fail validation because it should fail validation I think we're good.
Let me know if I missed something.

Reply:
Sorry I misread your previous comment, I thought you were saying query *could* be a string[] and I was asking you to add that to the types. But now I see you were saying typescript determines that it could be a string[], but we don't want it to be.